qos-monitor/src/main/java/eu/arrowhead/core/qos/dto/event/EventDTOConverter.java

```diff
@@ -0,0 +1,195 @@
+		Assert.isTrue(!Utilities.isEmpty(event.getPayload()), "Event payload is empty");
+		Assert.isTrue(!Utilities.isEmpty(event.getTimeStamp()), "Event timeStamp is empty");
+
+		final StartedMonitoringMeasurementEventDTO validEvent = new StartedMonitoringMeasurementEventDTO();
+		validEvent.setMetadata(event.getMetaData());
+		validEvent.setPayload(event.getPayload());
+		try {
+			validEvent.setTimeStamp(Utilities.parseUTCStringToLocalZonedDateTime(event.getTimeStamp()));
+		} catch (final DateTimeParseException ex) {
+
+			throw new InvalidParameterException("Event timestap has invalid format.", ex);
+		}
+
+		validateStartedMonitoringMeasurementEventDTOFields(validEvent);
+
+		return validEvent;
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	public static ReceivedMonitoringRequestEventDTO convertToReceivedMonitoringRequestEvent(final EventDTO event) {
+		logger.debug("convertToReceivedMonitoringRequestEvent started...");
+
+		Assert.notNull(event, "Event is null");
+		Assert.isTrue(!Utilities.isEmpty(event.getPayload()), "Event payload is empty");
+		Assert.isTrue(!Utilities.isEmpty(event.getTimeStamp()), "Event timeStamp is empty");
+
+		final ReceivedMonitoringRequestEventDTO validEvent = new ReceivedMonitoringRequestEventDTO();
+		validEvent.setMetadata(event.getMetaData());
+		validEvent.setPayload(event.getPayload());
+		try {
+			validEvent.setTimeStamp(Utilities.parseUTCStringToLocalZonedDateTime(event.getTimeStamp()));
+		} catch (final DateTimeParseException ex) {
+
+			throw new InvalidParameterException("Event timestap has invalid format.", ex);
+		}
+
+		validateReceivedMonitoringRequestEventDTOFields(validEvent);
+
+		return validEvent;
+	}
+
+	//=================================================================================================
+	// assistant methods
+
+	//-------------------------------------------------------------------------------------------------
+	private static void validateFinishedMonitoringMeasurementEventDTOFields(final FinishedMonitoringMeasurementEventDTO event) {
+
+		Assert.isTrue(event.getEventType().equals(QosMonitorEventType.FINISHED_MONITORING_MEASUREMENT), "Event type must be: FINISHED_MONITORING_MEASUREMENT");
+		Assert.isTrue(event.getMetadata().containsKey(QosMonitorConstants.PROCESS_ID_KEY), "Meta data must contain: " + QosMonitorConstants.PROCESS_ID_KEY);
+		Assert.isTrue(event.getMetadata().keySet().size() == QosMonitorConstants.FINISHED_MONITORING_MEASUREMENT_EVENT_PAYLOAD_METADATA_SIZE, "Meta data keys quantity is not valid");
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private static void validateInterruptedMonitoringMeasurementEventDTOFields(final InterruptedMonitoringMeasurementEventDTO event) {
+
+		Assert.isTrue(event.getEventType().equals(QosMonitorEventType.INTERUPTED_MONITORING_MEASUREMENT), "Event type must be: INTERUPTED_MONITORING_MEASUREMENT");
```

Comment:
typo in event type and in the message too: interupted => interrupted

Reply:
fixed